.github/workflows/publish.yaml

```diff
@@ -71,12 +71,21 @@ jobs:
         run: uds run dump-cluster-info
 
       - name: Publish Standard Package
+        if: ${{ github.event.inputs.snapshot != 'true' }}
         run: uds run -f tasks/publish.yaml standard-package --set FLAVOR=${{ matrix.flavor }}
 
       - name: Publish Upstream Flavored Bundles
-        if: ${{ matrix.flavor != 'registry1'}}
+        if: ${{ github.event.inputs.snapshot != 'true' && matrix.flavor != 'registry1' }}
         run: uds run -f tasks/publish.yaml bundles
 
+      - name: Publish Standard Package Snapshot
+        if: ${{ github.event.inputs.snapshot == 'true' }}
+        run: uds run -f tasks/publish.yaml standard-package --set FLAVOR=${{ matrix.flavor }} --set TARGET_REPO="ghcr.io/defenseunicorns/packages/uds/snapshots" --set VERSION="${SNAPSHOT_VERSION}"
+
+      - name: Publish Upstream Flavored Bundles Snapshot
+        if: ${{ github.event.inputs.snapshot == 'true' && matrix.flavor != 'registry1' }}
+        run: uds run -f tasks/publish.yaml bundles --set TARGET_REPO="ghcr.io/defenseunicorns/packages/uds/snapshots" --set VERSION="${SNAPSHOT_VERSION}"
```

Comment:
In my head was hoping these would be identical (not need separate one for snapshot) but for that to work I think we'd need to pass in release-please version + default repo (and lose out on the defaults in the tasks). Not sure which route is better in this case, either works - so feel free to resolve this based on your preference.

Reply:
@mjnagel i am leaning towards leaving as-is or perhaps we do the conditional inside the job step (setting the input to an env var) like this...

```yaml
      - name: Publish Standard Package
        run: |
          if [ "${SNAPSHOT}" = "true" ]; then
             uds run -f tasks/publish.yaml standard-package --set FLAVOR=${{ matrix.flavor }} --set TARGET_REPO="ghcr.io/mxnxpx/packages/uds/snapshots" --set VERSION="${SNAPSHOT_VERSION}"
          else
             uds run -f tasks/publish.yaml standard-package --set FLAVOR=${{ matrix.flavor }}
          fi

      - name: Publish Upstream Flavored Bundles
        if: ${{ matrix.flavor != 'registry1' }}
        run: |
          if [ "${SNAPSHOT}" = "true" ]; then
             uds run -f tasks/publish.yaml bundles --set TARGET_REPO="ghcr.io/mxnxpx/packages/uds/snapshots" --set VERSION="${SNAPSHOT_VERSION}"
          else
             uds run -f tasks/publish.yaml bundles
          fi
```

what do you think?

Reply:
My thought was more along the lines of:
```console
      - name: Publish Upstream Flavored Bundles
        if: ${{ matrix.flavor != 'registry1'}}
        run: uds run -f tasks/publish.yaml bundles --set TARGET_REPO="${{ inputs.repo }}" --set VERSION="${{ inputs.version }}"
```

And then passing inputs from each of the calling workflows. I think we can get the version from release-please outputs? And then hardcoding repo based on the calling workflow? Probably makes sense to remove the defaults from the tasks if we went this way.

Reply:
went down this path, but the changes needed seemed a bit harder to follow.  maybe we leave it like this for now and revisit later?

Reply:
Yep, as discussed offline going to press with current state and revisit for different refactoring in the future.